File: resources/algorithmic-toolbox/week2_algorithmic_warmup/2_last_digit_of_fibonacci_number/fibonacci_last_digit.cpp
#include <iostream>

int get_fibonacci_last_digit(int n) {
    n = n % 60; // This is because of Pisano Number of 10 is 60
    
    if (n <= 1)
        return n;

    int previous = 0;
    int current  = 1;

    for (int i = 0; i < n - 1; ++i) {
        int tmp_previous = previous;
        previous = current;
        current = tmp_previous + current;
    }

    return current % 10;
}

int main() {
    int n;
    std::cin >> n;
    
    int c = get_fibonacci_last_digit(n);
    std::cout << c << '\n';
}

File: resources/algorithmic-toolbox/week2_algorithmic_warmup/2_last_digit_of_fibonacci_number/fibonacci_last_digit.py
# Uses python3
import sys

def get_fibonacci_last_digit_naive(n):
    n = n % 60 # This is because of Pisano Number of 10 is 60
    
    if n <= 1:
        return n

    previous = 0
    current  = 1

    for _ in range(n - 1):
        previous, current = current, previous + current

    return current % 10

if __name__ == '__main__':
    input = sys.stdin.readline()
    n = int(input)
    print(get_fibonacci_last_digit_naive(n))
